libraries/shared/watcher/storage_watcher_test.go

```diff
@@ -51,401 +48,123 @@ var _ = Describe("Storage Watcher", func() {
 
 	Describe("Execute", func() {
 		var (
-			hashedAddress        common.Hash
 			storageWatcher       watcher.StorageWatcher
-			mockFetcher          *mocks.MockStorageFetcher
-			mockQueue            *mocks.MockStorageQueue
+			mockDiffsRepository  *mocks.MockStorageDiffRepository
 			mockHeaderRepository *fakes.MockHeaderRepository
-			mockTransformer      *mocks.MockStorageTransformer
 		)
 
 		BeforeEach(func() {
-			mockFetcher = mocks.NewMockStorageFetcher()
-			storageWatcher = watcher.NewStorageWatcher(mockFetcher, test_config.NewTestDB(test_config.NewTestNode()))
-
-			hashedAddress = types.HexToKeccak256Hash("0x0123456789abcdef")
-			mockTransformer = &mocks.MockStorageTransformer{KeccakOfAddress: hashedAddress}
-			storageWatcher.AddTransformers([]transformer.StorageTransformerInitializer{mockTransformer.FakeTransformerInitializer})
-
+			mockDiffsRepository = &mocks.MockStorageDiffRepository{}
 			mockHeaderRepository = &fakes.MockHeaderRepository{}
-			storageWatcher.HeaderRepository = mockHeaderRepository
+			storageWatcher = watcher.StorageWatcher{
+				HeaderRepository:          mockHeaderRepository,
+				StorageDiffRepository:     mockDiffsRepository,
+				KeccakAddressTransformers: map[common.Hash]transformer.StorageTransformer{},
+				RetryInterval:             time.Nanosecond,
+			}
 
-			mockQueue = &mocks.MockStorageQueue{}
-			storageWatcher.Queue = mockQueue
 		})
 
-		It("logs error if fetching storage diffs fails", func() {
-			mockFetcher.ErrsToReturn = []error{fakes.FakeError}
-			tempFile, fileErr := ioutil.TempFile("", "log")
-			Expect(fileErr).NotTo(HaveOccurred())
-			defer os.Remove(tempFile.Name())
-			logrus.SetOutput(tempFile)
-
-			err := storageWatcher.Execute(time.Hour)
-
-			Expect(err).To(MatchError(fakes.FakeError))
-			logContent, readErr := ioutil.ReadFile(tempFile.Name())
-			Expect(readErr).NotTo(HaveOccurred())
-			Expect(string(logContent)).To(ContainSubstring(fakes.FakeError.Error()))
+		It("marks diff as checked if no transformer is watching its address", func() {
+			unwatchedDiff := types.PersistedDiff{
+				RawDiff: types.RawDiff{
+					HashedAddress: test_data.FakeHash(),
+				},
+				ID: rand.Int63(),
+			}
+			mockDiffsRepository.GetNewDiffsDiffs = []types.PersistedDiff{unwatchedDiff}
+			mockDiffsRepository.GetNewDiffsErrors = []error{fakes.FakeError}
```

Comment:
Since Execute sets up an infinite loop that only dies if there's an error getting diffs from the db, all of these tests need to trigger that error to return. Would love to find a simpler setup, but the only alternative I could think of was firing the Execute calls in goroutines and making async assertions 🤢